packages/waas/src/intents/transactions.ts

```diff
@@ -59,6 +60,22 @@ export type SendERC1155Args = TransactionFeeArgs & {
   data?: string
 }
 
+export type SendContractCallArgs = TransactionFeeArgs & {
+  chainId: number
+  to: string
+  value?: ethers.BigNumberish
+  abi: string
+  func?: string
+  args: ContractCallArguments[]
+}
+
+export type ContractCallArguments = string | {
```

Comment:
"args" can either just be the argument as a string, or it can be another object which represents nested encoding to happen on server-side